src/test/java/com/codurance/lightaccess/PGOneToManyShould.java

```diff
@@ -0,0 +1,89 @@
+package com.codurance.lightaccess;
+
+import org.junit.Before;
+import org.junit.Test;
+import java.util.*;
+
+import static java.util.Arrays.*;
+import static java.util.Collections.*;
+import static java.util.Optional.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.core.IsEqual.equalTo;
+
+public class PGOneToManyShould {
+
+    private static final String KEY_WITHOUT_VALUES = "Key without value";
+    private static final String KEY_WITH_SINGLE_VALUE = "Key with single value";
+    private static final String KEY_WITH_MULTIPLE_VALUES = "Key with multiple value";
+    private static final String SOME_VALUE = "Value 1";
+    private static final String SOME_OTHER_VALUE = "Value 2";
+    private PGOneToMany<String, String> pgOneToMany;
+
+    @Before
+    public void setup() {
+        pgOneToMany = new PGOneToMany<>();
+    }
+
+    @Test
+    public void collectMappedValuesGroupedByKeyWhenKeyIsLinkedToMultipleValues() throws Exception {
+        pgOneToMany.put(new PGKeyValue<>(KEY_WITH_MULTIPLE_VALUES, of(SOME_VALUE)));
+        pgOneToMany.put(new PGKeyValue<>(KEY_WITH_MULTIPLE_VALUES, of(SOME_OTHER_VALUE)));
+        pgOneToMany.put(new PGKeyValue<>(KEY_WITH_SINGLE_VALUE, of(SOME_VALUE)));
+
+        assertThat(pgOneToMany.collect(MappedKeyValues::new),
+                containsInAnyOrder(
+                        new MappedKeyValues(KEY_WITH_MULTIPLE_VALUES, asList(SOME_VALUE, SOME_OTHER_VALUE)),
+                        new MappedKeyValues(KEY_WITH_SINGLE_VALUE, singletonList(SOME_VALUE))));
+    }
+
+    @Test
+    public void collectNoGroupedValuesWhenKeyIsNotLinkedToAnyValue() throws Exception {
+        pgOneToMany.put(new PGKeyValue<>(KEY_WITHOUT_VALUES, empty()));
+
+        assertThat(pgOneToMany.collect(MappedKeyValues::new),
+                equalTo(singletonList(new MappedKeyValues(KEY_WITHOUT_VALUES, emptyList()))));
+    }
+

```

Comment:
I can see another edge case when: You put one or more KeyValue pair with non-empty value and one or more KeyValue pair with an empty value.